lib/wallet/wallet.js

```diff
@@ -2569,6 +2570,157 @@ class Wallet extends EventEmitter {
     return mtx;
   }
 
+  async processNameForFinish(name, resource, acct) {
+    if (!rules.verifyName(name))
+      throw new Error(`Invalid name: "${name}".`);
+
+    const rawName = Buffer.from(name, 'ascii');
+    const nameHash = rules.hashName(rawName);
+    const ns = await this.getNameState(nameHash);
+    const height = this.wdb.height + 1;
+    const network = this.network;
+
+    if (!ns)
+      throw new Error(`Auction not found: "${name}".`);
+
+    if (ns.isExpired(height, network))
+      throw new Error(`Name has expired: "${name}"!`);
+
+    const state = ns.state(height, network);
+
+    if (state < states.CLOSED)
+      throw new Error(`Auction is not yet closed: "${name}".`);
+
+    const reveals = await this.txdb.getReveals(nameHash);
+    const outpoints = [];// { outpoint, output }
+
+    for (const {prevout, own} of reveals) {
+      const {hash, index} = prevout;
+
+      if (!own)
+        continue;
+
+      const coin = await this.getCoin(hash, index);
+
+      if (!coin)
+        continue;
+
+      if (acct != null && !await this.txdb.hasCoinByAccount(acct, hash, index))
+        continue;
+
+      // Is local?
+      if (coin.height < ns.height)
+        continue;
+
+      const output = new Output();
+      output.address = coin.address;
+      output.value = coin.value;
+      output.covenant.type = types.REDEEM;
+      output.covenant.pushHash(nameHash);
+      output.covenant.pushU32(ns.height);
+
+      if (prevout.equals(ns.owner)) { // Winner, does REGISTER
```

Comment:
When does ns.owner get set?

1) Are we sure that ns is not set on every tree interval, when the name tree gets committed? Or is it up to date with every block?
2) What happens if there is a redeem/register right on the exact close_block of an auction? Will `ns.owner` already point to the correct output?